packages/clay-time-picker/src/index.tsx

```diff
@@ -4,57 +4,422 @@
+					} else {
+						setValue(String((config as ConfigMaxMin).min));
+					}
+				} else {
+					setValue((config as ConfigAmpm).pm);
+				}
+				break;
+			case KEY_ARROWDOWN:
+				if (configName !== TimeType.ampm) {
+					if (value !== DEFAULT_VALUE && intrinsicValue !== 0) {
+						setValue(String(intrinsicValue - 1));
+					} else {
+						setValue(String((config as ConfigMaxMin).max));
+					}
+				} else {
+					setValue((config as ConfigAmpm).am);
+				}
+				break;
+			default:
+				if (regex.test(event.key) && configName !== TimeType.ampm) {
+					if (value === DEFAULT_VALUE) {
+						setValue(event.key);
+					} else {
+						setValue((value ? value : '') + event.key);
+					}
+				}
+				break;
+		}
+	};
+
+	const handleAction = (direction: number) => {
+		const {configName} = elementFocusedRef.current;
+		const config = useConfig[configName];
+		const prevValue = values[configName];
+		const value =
+			configName === TimeType.ampm
+				? direction === 1
+					? (config as ConfigAmpm).pm
+					: (config as ConfigAmpm).am
+				: handleMaxAndMin(
+						String(
+							(prevValue !== DEFAULT_VALUE
+								? Number(prevValue)
+								: 0) + direction
+						),
+						config as ConfigMaxMin
+				  );
 
-	return timePickerContent(otherProps);
+		onInputChange({
+			...values,
+			// eslint-disable-next-line sort-keys
+			[configName]: value,
+		});
+	};
+
+	const handleDocumentClick = (event: Event) => {
+		if (
+			elementRef.current &&
+			event.target !== null &&
+			!elementRef.current.contains(event.target as HTMLDivElement)
+		) {
+			elementFocusedRef.current = defaultElementFocus;
+			setActionVisible(false);
+			setIsFocused(false);
+		}
+	};
+
+	useEffect(() => {
+		document.addEventListener('click', handleDocumentClick);
+
+		return () => {
+			document.removeEventListener('click', handleDocumentClick);
+		};
+	}, []);
+
+	const visibleActionReset =
+		actionVisible &&
+		(values.ampm !== DEFAULT_VALUE ||
+			values.hours !== DEFAULT_VALUE ||
+			values.minutes !== DEFAULT_VALUE);
+
+	return (
+		<div className="clay-time">
+			<div className="input-group">
+				{icon && (
+					<div className="input-group-item input-group-item-shrink">
+						<div className="input-group-text">
+							<ClayIcon spritemap={spritemap} symbol="time" />
+						</div>
+					</div>
+				)}
+				<div className="input-group-item input-group-item-shrink">
+					<div
+						className={classNames('form-control', {
+							focus: isFocused,
+						})}
+						onMouseEnter={() => setActionVisible(true)}
+						onMouseLeave={() => {
+							if (!elementFocusedRef.current.focused) {
+								setActionVisible(false);
+								setIsFocused(false);
+							}
+						}}
+						ref={elementRef}
+					>
+						<div className="clay-time-edit">
+							<input
+								className="clay-time-hours form-control-inset"
+								id="timePickerHours"
+								maxLength={2}
+								name="hours"
+								onFocus={() => {
+									elementFocusedRef.current = {
```

Comment:
since this ref isnt actually an element, what do you think about calling it just `focusedRef`?